packages/client/hooks/useTransition.ts

```diff
@@ -76,17 +76,17 @@ const useTransition = <T extends {key: Key}>(children: T[]) => {
     const {current: prevTChildren} = previousTransitionChildrenRef
 
     let touched = false
-    // add mounted nodes
+    // add mounted nodes + update new orderings
     const mountingKeys = [] as Key[]
-    children.forEach((nextChild) => {
+    children.forEach((nextChild, idxInNext) => {
       const idxInPrev = prevTChildren.findIndex(({child}) => child.key === nextChild.key)
       const status = idxInPrev === -1 ? TransitionStatus.MOUNTED : prevTChildren[idxInPrev]!.status
       currentTChildren.push({
         status,
         child: nextChild,
         onTransitionEnd: transitionEndFactory(nextChild.key)
       })
-      if (idxInPrev === -1) {
+      if (idxInPrev === -1 || idxInPrev !== idxInNext) {
         touched = true
         mountingKeys.push(nextChild.key)
```

Comment:
+1 since `mountingKeys` contains now all changed keys, it would be nice to change the name accordingly

Reply:
Changed to `updatedKeys`